autoload/tagbar.vim

```diff
@@ -2866,6 +2866,46 @@ function! s:EscapeCtagsCmd(ctags_bin, args, ...) abort
     return ctags_cmd
 endfunction
 
+" run shell command in a proper way: prevent temporary window creation
+function! s:run_system(cmd, version) abort
+    if has('nvim')
+        let hr = system(a:cmd)
+    if has('win32') || has('win64') || has('win95') || has('win16')
+        if a:version <= 0 || (has('python3') == 0 && has('python2') == 0)
+            let hr = system(a:cmd)
+            let s:shell_error = v:shell_error
+            return hr
+        elseif a:version == 3
+            let pyx = 'py3 '
+            let python_eval = 'py3eval'
+        elseif a:version == 2
+            let pyx = 'py2 '
+            let python_eval = 'pyeval'
+        else
```

Comment:
What is this `pyxeval` thing? What are we even targeting here?

Reply:
the purpose of `a:version` is for user to indicate python version:

- 0: disable this feature
- 1: automatically choose python version, by ultilizing `pyx` and `pyxeval` commands.
- 2: force use python 2 
- 3: force use python 3

Both `pyx` and `pyxeval` are internal vim commands to perform `py`/`pyeval` or `py3`/`py3eval` automatically.

Reply:
Nevermind I didn't realize that cal a call() thing, carry on...